app/models/game.rb

```diff
@@ -1,5 +1,9 @@
 class Game < ApplicationRecord
-  belongs_to :white_player, class_name: 'Player'
-  belongs_to :black_player, class_name: 'Player'
+  belongs_to :white_player, class_name: 'User'
+  belongs_to :black_player, class_name: 'User', required: false
   has_many :pieces
+
+  def self.needs_second_player?
```

Comment:
One small suggestion - this method will return a listing of all games that match the `where` condition, not true or false, so you may want to remove the `?` from the method name.